src/commands/synthetics/__tests__/cli.test.ts

```diff
@@ -1047,80 +1080,155 @@ describe('run-test', () => {
-        })
-      )
+        const testOverrideEnv = {
+          DATADOG_SYNTHETICS_OVERRIDE_ALLOW_INSECURE_CERTIFICATES: 'true',
+          DATADOG_SYNTHETICS_OVERRIDE_BASIC_AUTH_PASSWORD: 'password-from-env',
+          DATADOG_SYNTHETICS_OVERRIDE_BASIC_AUTH_USERNAME: 'username-from-env',
+          DATADOG_SYNTHETICS_OVERRIDE_BODY: 'body-from-env',
+          DATADOG_SYNTHETICS_OVERRIDE_BODY_TYPE: 'bodyType-from-env',
+          DATADOG_SYNTHETICS_OVERRIDE_COOKIES: 'cookie1-from-env;cookie2-from-env;cookie3-from-env',
+          DATADOG_SYNTHETICS_OVERRIDE_COOKIES_APPEND: 'true',
+          DATADOG_SYNTHETICS_OVERRIDE_DEFAULT_STEP_TIMEOUT: '42',
+          DATADOG_SYNTHETICS_OVERRIDE_DEVICE_IDS: 'chrome.laptop_large_from_env',
+          DATADOG_SYNTHETICS_OVERRIDE_EXECUTION_RULE: 'BLOCKING',
+          DATADOG_SYNTHETICS_OVERRIDE_FOLLOW_REDIRECTS: 'true',
+          DATADOG_SYNTHETICS_OVERRIDE_HEADERS:
+            "{'Content-Type': 'application/json', 'Authorization': 'Bearer token from env'}",
+          // TODO SYNTH-12989: Clean up `locations` that should only be part of the testOverrides
+          DATADOG_SYNTHETICS_LOCATIONS: 'Wonderland;FarFarAway',
+          DATADOG_SYNTHETICS_OVERRIDE_LOCATIONS: 'location_1_from_env;location_2_from_env',
+          DATADOG_SYNTHETICS_OVERRIDE_MOBILE_APPLICATION_VERSION: 'env-00000000-0000-0000-0000-000000000000',
+          DATADOG_SYNTHETICS_OVERRIDE_RESOURCE_URL_SUBSTITUTION_REGEXES: 'env-regex1;env-regex2',
+          DATADOG_SYNTHETICS_OVERRIDE_RETRY_COUNT: '5',
+          DATADOG_SYNTHETICS_OVERRIDE_RETRY_INTERVAL: '100',
+          DATADOG_SYNTHETICS_OVERRIDE_START_URL: 'startUrl-from-env',
+          DATADOG_SYNTHETICS_OVERRIDE_START_URL_SUBSTITUTION_REGEX: 'startUrlSubstitutionRegex-from-env',
+          DATADOG_SYNTHETICS_OVERRIDE_TEST_TIMEOUT: '42',
+          DATADOG_SYNTHETICS_OVERRIDE_VARIABLES: "{'envVar1': 'value1', 'envVar2': 'value2'}",
+        }
 
-      // Global < ENV
-      process.env = {
-        DATADOG_SYNTHETICS_LOCATIONS: 'aws:us-east-3',
-      }
-      expect(await command.execute()).toBe(0)
-      expect(getTestsToTriggerMock).toHaveBeenNthCalledWith(
-        2,
-        ...getExpectedTestsToTriggerArguments({
-          locations: ['aws:us-east-3'],
-          mobileApplicationVersionFilePath: './path/to/application_global.apk',
-          pollingTimeout: DEFAULT_POLLING_TIMEOUT,
-        })
-      )
-      // Same, but with 2 locations.
-      process.env = {
-        DATADOG_SYNTHETICS_LOCATIONS: 'aws:us-east-3;aws:us-east-4',
-      }
-      expect(await command.execute()).toBe(0)
-      expect(getTestsToTriggerMock).toHaveBeenNthCalledWith(
-        3,
-        ...getExpectedTestsToTriggerArguments({
-          locations: ['aws:us-east-3', 'aws:us-east-4'],
-          mobileApplicationVersionFilePath: './path/to/application_global.apk',
-          pollingTimeout: DEFAULT_POLLING_TIMEOUT,
-        })
-      )
+        process.env = testOverrideEnv
```

Comment:
Maybe let's not reassign `process.env` to avoid removing any existing unrelated env variable?
```suggestion
        Object.assign(process.env, testOverrideEnv)
```

Reply:
I'm not sure, we do this on multiple occurrences in other tests (on line 102, 163,  708, 949) and it seems fine... And for the scope of the test, I guess we actually want to only have those variables and remove the unrelated ones?